images/executor/src/index.ts

```diff
@@ -0,0 +1,112 @@
+import readline from "readline";
+import path from "path";
+import {
+  PythonCaller,
+  CallerOptions,
+  FunctionMissingError
+} from "./code-caller";
+
+interface Request {
+  file: string;
+  fcn: string;
+  args: any[];
+  cwd: string;
+  paths: string[];
+}
+
+interface Results {
+  error?: string;
+  data?: any;
+  output?: string;
+  functionMissing?: boolean;
+  needsFullRestart: boolean;
+}
+
+async function handleInput(
+  line: string,
+  caller: PythonCaller
+): Promise<Results> {
+  return new Promise((resolve, reject) => {
```

Comment:
Why do we explicitly return a Promise, rather than just using async/await code in this function?

Reply:
This code no longer exists.